src/backend/expungeservice/endpoints/search.py

```diff
@@ -0,0 +1,58 @@
+from flask.views import MethodView
+from flask import request, current_app
+
+from expungeservice.request import check_data_fields
+from expungeservice.request.error import error
+from expungeservice.crawler.crawler import Crawler
+from expungeservice.expunger.expunger import Expunger
+from expungeservice.endpoints.auth import user_auth_required
+from expungeservice.serializer import ExpungeModelEncoder
+from expungeservice.crypto import DataCipher
+
+
+class Search(MethodView):
+
+    @user_auth_required
+    def post(self):
+        request_data = request.get_json()
+
+        if request_data is None:
+            error(400, "No json data in request body")
+
+        check_data_fields(request_data, ["first_name", "last_name",
+                                 "middle_name", "birth_date"])
+
+        cipher = DataCipher(
+            key=current_app.config.get("JWT_SECRET_KEY"))
+
+        decrypted_credentials = cipher.decrypt(request.cookies["oeci_token"])
+
+        crawler = Crawler()
+
+        login_result = crawler.login(
+            decrypted_credentials["oeci_username"],
+            decrypted_credentials["oeci_password"],
+            close_session=False)
+
```

Comment:
We should check if the login was successful before attempting to do search. I would advise against assuming the login was successful as there are various reasons it could fail even with the correct username and password. For example: The e-courts site could be taken down for maintenance, or it could be offline for other reasons.